python/tank/hook.py

```diff
@@ -631,17 +648,22 @@ def create_hook_instance(hook_paths, parent):
 
         3. /tmp/c.py is finally loaded in, get_hook_baseclass() now returns HookB.
 
-        4. HookC class is instantiated and method method_name is executed.
+        4. HookC class is returned.
+
+    An optional `base_class` can be provided to override the default ``Hook``
+    base class. This is useful for bundles that create hook instances at
+    execution time and wish to provide default implementation without the need
+    to configure the base hook.
 
     :param hook_paths: List of full paths to hooks, in inheritance order.
-    :param parent: Parent object. This will be accessible inside
-                   the hook as self.parent, and is typically an
-                   app, engine or core object.
+    :param base_class: A python class to use as the base class for the created
+        hook. This will override the default hook base class, ``Hook``.
     :returns: Instance of the hook.
     """
+
     # keep track of the current base class - this is used when loading hooks to dynamically
     # inherit from the correct base.
-    _current_hook_baseclass.value = Hook
+    _current_hook_baseclass.value = base_class or Hook
```

Comment:
Worth making sure base_class is a descendent of Hook?

Reply:
done.